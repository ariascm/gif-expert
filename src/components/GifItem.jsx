
export const GifItem = ({ title, url }) => {

    // console.log(a)
    // console.log(title)

    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
