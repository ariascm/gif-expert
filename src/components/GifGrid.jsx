
import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"
import PropTypes from "prop-types"


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category)



    return (
        <>
            <h3>{category}</h3>
            {isLoading && (<h4>cargando...</h4>)}


            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem
                            key={image.id}
                            {...image}  //esparcimos y enviamos todas las propiedades de imagen (title, id, url)
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
category: PropTypes.string.isRequired
}