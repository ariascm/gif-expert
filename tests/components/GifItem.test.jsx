import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en GifItem', () => {

    const title = "The batman"
    const url = "https://thebatman.com/batman.jpg"

    test('Pruebas en <GifItem/>', () => {

        const { container } = render(<GifItem title={title} url={url} />)

        expect(container).toMatchSnapshot()
    })


    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url} />)

        const { src, alt } = screen.getByRole('img')

        expect(src).toBe(url)
        expect(alt).toBe(title)

    })

    test('debe mostrar el texto del titulo', () => {

        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()

    })
})