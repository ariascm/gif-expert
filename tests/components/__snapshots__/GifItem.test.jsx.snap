// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Pruebas en <GifItem/> 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="The batman"
      src="https://thebatman.com/batman.jpg"
    />
    <p>
      The batman
    </p>
  </div>
</div>
`;
